server/src/main/java/org/opensearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java

```diff
@@ -152,7 +151,7 @@ public RestoreSnapshotRequest(StreamInput in) throws IOException {
         if (in.getVersion().onOrAfter(Version.V_2_7_0)) {
             storageType = in.readEnum(StorageType.class);
         }
-        if (FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE) && in.getVersion().onOrAfter(Version.V_2_9_0)) {
+        if (in.getVersion().onOrAfter(Version.CURRENT)) {
```

Comment:
Shouldn't it still be `if (in.getVersion().onOrAfter(Version. V_2_9_0)) {`

Reply:
+1

Reply:
That will end up breaking BwC. Issue created here: https://github.com/opensearch-project/OpenSearch/issues/9792

Reply:
then lets use `2.10` instead ?

Reply:
Should we have something like this -
```
if (in.getVersion().onOrAfter(Version.CURRENT) || (indexSettings.isRemoteEnabled() && in.getVersion().onOrAfter(Version.V_2_9_0))) {
            sourceRemoteStoreRepository = in.readOptionalString();
        }
```
isRemoteEnabled is for illustration, we will need to check the specific setting in indexSettings?

Reply:
We won't get the index settings here. IMO we should not Ser-De any fields in the API depending on a specific feature to maintain backwards compatibility for this API. We should just read/write the field to the transport layer as is and use it in the actual functionality with specific clauses.

Reply:
Changed to `Version.V_3_0_0` in latest commit